projects/core/src/multi-location-initialized/multi-location-initialized.module.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { LOCATION_INITIALIZED } from '@angular/common';
+import { ModuleWithProviders, NgModule, inject } from '@angular/core';
+import { MULTI_LOCATION_INITIALIZED } from './multi-location-initialized-token';
+
+/**
+ * Factory function that initializes the location.
+ * It retrieves the initializers from the MULTI_LOCATION_INITIALIZED injection token,
+ * invokes each initializer, and returns a promise that resolves when all initializers have completed.
+ * @returns A promise that resolves when all initializers have completed.
+ */
+function locationInitializedFactory() {
+  const initializers =
+    inject(MULTI_LOCATION_INITIALIZED, { optional: true }) ?? [];
+  const promiseInitializers = initializers.map((initializer) => initializer());
+  return Promise.all(promiseInitializers);
+}
+
+@NgModule()
+export class MultiLocationInitializedModule {
```

Comment:
I don't think we need a new module for it, especially with this name.
What do you think about defining the LOCATION_INITIALIZED hook in the core `RoutingModule` ?